x-pack/plugins/security_solution/server/lib/detection_engine/signals/threshold_get_bucket_filters.ts

```diff
@@ -74,8 +74,9 @@ export const getThresholdBucketFilters = async ({
         if (signalTerms == null) {
           signalTerms = [
             {
-              field: (((hit._source.rule as RulesSchema).threshold as unknown) as { field: string })
-                .field,
+              field: (((hit._source.signal?.rule as RulesSchema).threshold as unknown) as {
+                field: string;
+              }).field,
```

Comment:
might be easier to use lodash `get` here and then check if we got the value successfully rather than triple casting

Reply:
You don't like my triple cast? 😂

Reply:
Propose we leave this for now and I can update in a follow-up... just want to focus on getting the fixes in for BC3.